packages/jest-cli/src/TestRunner.js

```diff
@@ -92,95 +84,36 @@ class TestRunner {
     this._dispatcher.unregister(ReporterClass);
   }
 
-  _getTestPerformanceCachePath() {
-    const config = this._config;
-    return getCacheFilePath(config.cacheDirectory, 'perf-cache-' + config.name);
-  }
-
-  _sortTests(testPaths: Array<string>) {
-    // When running more tests than we have workers available, sort the tests
-    // by size - big test files usually take longer to complete, so we run
-    // them first in an effort to minimize worker idle time at the end of a
-    // long test run.
-    //
-    // After a test run we store the time it took to run a test and on
-    // subsequent runs we use that to run the slowest tests first, yielding the
-    // fastest results.
-    try {
-      if (this._config.cache) {
-        this._testPerformanceCache = JSON.parse(
-          fs.readFileSync(this._getTestPerformanceCachePath(), 'utf8'),
-        );
-      } else {
-        this._testPerformanceCache = {};
-      }
-    } catch (e) {
-      this._testPerformanceCache = {};
-    }
-
-    const cache = this._testPerformanceCache;
+  async runTests(tests: Tests, watcher: TestWatcher) {
     const timings = [];
-    const stats = {};
-    const getFileSize = filePath =>
-      stats[filePath] || (stats[filePath] = fs.statSync(filePath).size);
-    const getTestRunTime = filePath => {
-      if (cache[filePath]) {
-        return cache[filePath][0] === FAIL ? Infinity : cache[filePath][1];
-      }
-      return null;
-    };
-
-    testPaths = testPaths.sort((pathA, pathB) => {
-      const timeA = getTestRunTime(pathA);
-      const timeB = getTestRunTime(pathB);
-      if (timeA != null && timeB != null) {
-        return timeA < timeB ? 1 : -1;
-      }
-      return getFileSize(pathA) < getFileSize(pathB) ? 1 : -1;
-    });
-
-    testPaths.forEach(filePath => {
-      const timing = cache[filePath] && cache[filePath][1];
-      if (timing) {
-        timings.push(timing);
+    tests.forEach(test => {
```

Comment:
Glad you are bringing this up! I actually can't simplify because flow isn't smart enough to realize that the filter call filters out falsy values. It types `timings` as `Array<?number>` instead of `Array<number>` :(